### PR TITLE
fix: make `hide-shorts-button` fingerprint more future-proof

### PR DESCRIPTION
**Pull request by ponces**
This fixes an error when applying this patch on v17.28.34 (and most likely other future versions) while still working with versions mentioned on the compatibility class.

### CODE REVIEW

**Comment by TheJeterLP** [comment]
17.28 is a beta. We should wait until its a full release.

**Comment by ponces** [comment]
I missed changing the index on the replace instruction :o
One sec!

**Comment by ponces** [comment]
> 17.28 is a beta. We should wait until its a full release.

Is it? I don't think 17.28.34 is a beta. I receive the update through Play Store (not on beta program) and based on APK Mirror the beta version is in fact 17.28.32, not 17.28.34...

**Comment by TheJeterLP** [comment]
> > 17.28 is a beta. We should wait until its a full release.
> 
> Is it? I don't think 17.28.34 is a beta. I receive the update through Play Store (not on beta program) and based on APK Mirror the beta version is in fact 17.28.32, not 17.28.34...

https://www.apkmirror.com/apk/google-inc/youtube/
look at the release dates. Latest released stable is 17.27.39, and thats the one I receive from playstore

**Comment by oSumAtrIX** [comment]
This PR does not consider updating the compatiblility annotation. This means the FP has to be compatible with the current versions.

**Comment by ponces** [comment]
> This PR does not consider updating the compatiblility annotation. This means the FP has to be compatible with the current versions.

I'll update it once I have the indexes sorted out too. This can be considered as a draft for now.

**Comment by ponces** [comment]
> > > 17.28 is a beta. We should wait until its a full release.
> > 
> > Is it? I don't think 17.28.34 is a beta. I receive the update through Play Store (not on beta program) and based on APK Mirror the beta version is in fact 17.28.32, not 17.28.34...
> 
> https://www.apkmirror.com/apk/google-inc/youtube/
> look at the release dates. Latest released stable is 17.27.39, and thats the one I receive from playstore

Still...
It doesn't mean it's a beta.
Play Store rollouts are all sorts of weird stuff!

**Comment by TheJeterLP** [comment]
Weird that 17.27.39 is newer then

File: src/main/kotlin/app/revanced/patches/youtube/layout/shorts/button/annotations/ShortsButtonCompatibility.kt [hunk 1]
```diff
@@ -5,7 +5,7 @@ import app.revanced.patcher.annotation.Package
 
 @Compatibility(
     [Package(
-        "com.google.android.youtube", arrayOf("17.14.35", "17.17.34", "17.19.36", "17.20.37", "17.22.36", "17.23.35", "17.23.36", "17.24.34", "17.24.35", "17.25.34", "17.26.35", "17.27.39")
+        "com.google.android.youtube", arrayOf("17.14.35", "17.17.34", "17.19.36", "17.20.37", "17.22.36", "17.23.35", "17.23.36", "17.24.34", "17.24.35", "17.25.34", "17.26.35", "17.27.39", "17.28.34")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/shorts/button/annotations/ShortsButtonCompatibility.kt:8]: You added a new version, but is the fingerprint still compatible with the previous one as you changed it?

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

File: src/main/kotlin/app/revanced/patches/youtube/layout/shorts/button/annotations/ShortsButtonCompatibility.kt (hunk 1 above)
Comment (ponces) [reply src/main/kotlin/app/revanced/patches/youtube/layout/shorts/button/annotations/ShortsButtonCompatibility.kt:8]: Yes, that was point of this PR. Check my first comment on this thread.

**Comment by ponces** [comment]
It's working with the latest push.

**Comment by TheJeterLP** [comment]
Could you update and test the other patches too then?

**Comment by ponces** [comment]
> Could you update and test the other patches too then?

Could you please point to which patches you're referring to?

**Comment by TheJeterLP** [comment]
> > Could you update and test the other patches too then?
> 
> Could you please point to which patches you're referring to?

all of them

**Comment by ponces** [comment]
> > > Could you update and test the other patches too then?
> > 
> > Could you please point to which patches you're referring to?
> 
> all of them

You mean the compatibility list?
Because the ones that I'm using are all correctly being applied.
I can't confirm those which I'm excluding: sponsorblock, custom-branding, premium-heading , hide-cast-button and microg-support.

**Comment by TheJeterLP** [comment]
> > > > Could you update and test the other patches too then?
> > > 
> > > 
> > > Could you please point to which patches you're referring to?
> > 
> > 
> > all of them
> 
> You mean the compatibility list? Because the ones that I'm using are all correctly being applied. I can't confirm those which I'm excluding: sponsorblock, custom-branding, premium-heading, hide-cast-button and microg-support.

Okay, I'll do it then

**Comment by ponces** [comment]
> > > > > Could you update and test the other patches too then?
> > > > 
> > > > 
> > > > Could you please point to which patches you're referring to?
> > > 
> > > 
> > > all of them
> > 
> > You mean the compatibility list? Because the ones that I'm using are all correctly being applied. I can't confirm those which I'm excluding: sponsorblock, custom-branding, premium-heading, hide-cast-button and microg-support.
> 
> Okay, I'll do it then

I can do it, no worries. It's just I'm not on my laptop right now so I can't confirm the others...

**Comment by ponces** [comment]
> > > Could you update and test the other patches too then?
> > 
> > 
> > Could you please point to which patches you're referring to?
> 
> all of them

Is [this](https://github.com/revanced/revanced-patches/pull/225) ok?

**Comment by TheJeterLP** [comment]
> > > > Could you update and test the other patches too then?
> > > 
> > > 
> > > Could you please point to which patches you're referring to?
> > 
> > 
> > all of them
> 
> Is [this](https://github.com/revanced/revanced-patches/pull/225) ok?

yes, awesome thank you very much!

**Pull request merged**